frontend/device/generic/device.lua

```diff
@@ -144,6 +144,31 @@ function Device:onPowerEvent(ev)
         self.screen_saver_mode = false
         self.powerd:refreshCapacity()
         self.powerd:afterResume()
+    elseif ev == "PowerLong" and not self.screen_saver_mode then
+        local UIManager = require("ui/uimanager")
+        local ConfirmBox = require("ui/widget/confirmbox")
+        local InfoMessage = require("ui/widget/infomessage")
+        local ReaderUI = require("apps/reader/readerui")
+        local FileManager = require("apps/filemanager/filemanager")
```

Comment:
Requiring these heavy modules (especially `ReaderUI` and `FileManager`) makes the low-level generic device coupling with the high-level applications which is not a good idea. How about just sending a `Close` event and let the `ReaderUI` and `FileManager` handle the `Close` event themselves. This should be OK:
```
UIManager:sendEvent(Event:new("Close"))
```
Similar use case can be found at `Device:onPowerEvent` on handling `Power` and `Suspend` events.